### PR TITLE
[Task.45593] Lam filter search va sort

### PR DESCRIPTION
**Pull request by ghost**
https://edu-redmine.sun-asterisk.vn/issues/45593

image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/Component/Shop/Search/index.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }, 300);
+
+  function handleSearchTermChange(e) {
+    const inputValue = e.target.value;
+    dispatch(search(inputValue));
+    debounceOnChange(inputValue);
+  }
+
+  return (
+    <div className="search">
```
Comment (vietnq-1097) [inline src/Component/Shop/Search/index.jsx]:
```suggestion
    <form className="search">
```
input thì cho vào form nhé

File: src/Component/Shop/Search/index.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <input
+        type="text"
+        className="form-control"
+        placeholder="Search a product"
+        value={searchTerm}
+        onChange={handleSearchTermChange}
+      />
+      <button className="search-btn">
```
Comment (vietnq-1097) [inline src/Component/Shop/Search/index.jsx]:
```suggestion
      <button type="submit" className="search-btn">
```

File: src/Component/Shop/Sort/index.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    } else {
+      dispatch(
+        setProductsShopFilter({
+          ...filter,
+          _order: value,
+          _sort: "price",
+        })
+      );
+      dispatch(setSelected(value));
+    }
```
Comment (vietnq-1097) [inline src/Component/Shop/Sort/index.jsx]:
```suggestion
   const selectedValue = value === DEFAULT_OPTIONS ? "" : value;
   dispatch(
     setProductsShopFilter({
       ...filter,
       _order: selectedValue,
        _sort: value === DEFAULT_OPTIONS ? "" : "price",
      })
    );
    dispatch(setSelected(selectedValue));
```

**Comment by ghost** [comment]
ready

**Pull request merged**